contracts/p1/mixins/RecollateralizationLib.sol

```diff
@@ -114,172 +106,120 @@ library RecollateralizationLibP1 {
+            }
+
+            // Ignore dust amounts for assets not in the basket; their value is inaccessible
+            // {tok} = {tok/BU} * {BU}
+            uint192 inBasket = ctx.bh.quantity(reg.erc20s[i]).mul(ctx.basketsHeld, FLOOR);
+            if (bal < inBasket) inBasket = bal; // not sure if needed
+
+            // Skip over dust-balance assets not in the basket
+            {
+                (uint192 lotLow, ) = reg.assets[i].lotPrice(); // {UoA/tok}
+
+                // Intentionally include value of IFFY/DISABLED collateral
+                if (
+                    inBasket == 0 &&
+                    !TradeLib.isEnoughToSell(reg.assets[i], bal, lotLow, ctx.minTradeVolume)
+                ) continue;
             }
 
             (uint192 low, uint192 high) = reg.assets[i].price(); // {UoA/tok}
-            (uint192 lotLow, ) = reg.assets[i].lotPrice(); // {UoA/tok}
 
-            // Ignore dust amounts for assets not in the basket; their value is inaccessible
-            if (
-                components.bh.quantity(reg.erc20s[i]) == 0 &&
-                !TradeLib.isEnoughToSell(reg.assets[i], bal, lotLow, rules.minTradeVolume)
-            ) continue;
-
-            // Intentionally include value of IFFY/DISABLED collateral when low is nonzero
-            // {UoA} = {UoA} + {UoA/tok} * {tok}
-            assetsLow += low.mul(bal, FLOOR);
-            // += is same as Fix.plus
-
-            // assetsHigh += high.mul(bal, CEIL), where assetsHigh is [0, FIX_MAX]
-            // {UoA} = {UoA/tok} * {tok}
-            uint192 val = components.bm.safeMulDivCeil(high, bal, FIX_ONE);
-            if (uint256(assetsHigh) + val >= FIX_MAX) assetsHigh = FIX_MAX;
-            else assetsHigh += val;
-            // += is same as Fix.plus
-
-            // Accumulate potential losses to dust
-            potentialDustLoss = potentialDustLoss.plus(rules.minTradeVolume);
+            assert(high != FIX_MAX || inBasket == 0); // collateral in the basket must be priced
+
+            // throughout this section +/- is same as Fix.plus/Fix.minus
+
+            // range.top: contribution from balance beyond basketsHeld
+            {
+                // sell at high and buy at low
+                // needs overflow protection: unpriced asset with price [0, FIX_MAX] can overflow
+                // {BU} = {UoA/tok} * {tok} / {UoA/BU}
+                uint192 b = ctx.bm.safeMulDivCeil(high, bal - inBasket, basketPriceLow);
+                if (uint256(range.top) + b >= FIX_MAX) range.top = FIX_MAX;
+                else range.top += b;
+            }
+
+            // range.bottom: contribution from balance beyond basketsHeld
+            {
+                // sell at low and buy at high
+                // {UoA} = {UoA/tok} * {tok}
+                uint192 b = low.mul(bal - inBasket, FLOOR);
+
+                // Account for potential dust loss
+                b = (b < ctx.minTradeVolume) ? 0 : b - ctx.minTradeVolume;
```

Comment:
It's because in the worst-case any auction could fill for the entire volume minus `minTradeVolume`. It might return `minTradeVolume` worth of sell tokens to us, and since we won't sell that amount, it becomes lost value. We have to build this into our projection of `range.bottom` ahead of time. 

More formally: for each token we sell, the maximum of `minTradeVolume` and the UoA value of the amount being sold might become inaccessible as dust.